res/controllers/midi-components-0.0.js

```diff
@@ -311,45 +308,49 @@
         shift: function() {
             this.inKey = "hotcue_" + this.number + "_clear";
         },
-        getColor: function() {
-            if (this.colorIdKey !== undefined) {
-                return color.predefinedColorFromId(engine.getValue(this.group, this.colorIdKey));
-            } else {
-                return null;
+        getColor: function(colorCode) {
+            if (colorCode === undefined && this.colorKey !== undefined) {
+                colorCode = engine.getValue(this.group, this.colorKey);
+            }
+
+            if (colorCode !== undefined) {
+                return {
+                    "red": (colorCode >> 16) & 0xFF,
```

Comment:
Let's move this bit shifting to convert from the CO value to an RGB JS object into a function into common-controller-scripts.js

Reply:
Another function in common-controller-scripts.js to do the reverse for setting the color CO may be useful too.

Reply:
Done. https://github.com/mixxxdj/mixxx/pull/2520/commits/702050f0b8c52622849d0db1fa5146589c52d11f